### ISSUE TITLE
Improve S3863 by only reporting duplicated imports of the same kind

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#2458]
Community: https://community.sonarsource.com/t/typescript-import-type-and-value-from-same-module/35776

--
Since we use ESLint's `no-duplicate-imports`, the rule does not take into account the two types of imports supported by TypeScript, namely _value import_ and _type import_. This leads to false positive when one imports types and values from the same module:

```typescript
import type { Foo } from './module';
import { Bar } from './module';
```

**Comment by Vampire2008** [comment]
There is TypeScript variant of this rule [@typescript-eslint/no-duplicate-imports](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-duplicate-imports.md)

**Comment by yassin-kammoun-sonarsource** [comment]
Thanks for sharing!

**Comment by yassin-kammoun-sonarsource** [comment]
Not sure what happened here, but the rule already made the distinction between type imports and value imports. No issue is raised on the provided code snippet with latest SonarQube.
